clearnode/app_session_service.go

```diff
@@ -339,8 +354,195 @@ func (s *AppSessionService) GetAppSessions(participantWallet string, status stri
+			}
+
+			if err := ledger.Record(userAccountID, alloc.AssetSymbol, depositAmount.Neg()); err != nil {
+				return nil, RPCErrorf("failed to debit source account: %w", err)
+			}
+			if err := ledger.Record(sessionAccountID, alloc.AssetSymbol, depositAmount); err != nil {
+				return nil, RPCErrorf("failed to credit destination account: %w", err)
+			}
+			_, err = RecordLedgerTransaction(tx, TransactionTypeAppDeposit, userAccountID, sessionAccountID, alloc.AssetSymbol, depositAmount)
+			if err != nil {
+				return nil, RPCErrorf("failed to record transaction: %w", err)
+			}
+
+			participants[walletAddress] = true
+		} else if alloc.Amount.LessThan(currentAmount) {
+			return nil, RPCErrorf("incorrect deposit request: decreased allocation for participant %s", walletAddress)
+		}
+	}
+
+	if noDeposits {
+		return nil, RPCErrorf("incorrect deposit request: non-positive allocations sum delta")
+	}
+
+	return participants, nil
+}
+
+func (s *AppSessionService) handleWithdrawIntent(tx *gorm.DB, appSession AppSession, params *SubmitAppStateParams, sessionAccountID AccountID) (map[string]bool, error) {
+	participants := make(map[string]bool)
+
+	currentAllocations, err := getParticipantAllocations(tx, appSession, sessionAccountID)
+	if err != nil {
+		return nil, RPCErrorf("failed to get current allocations: %w", err)
+	}
+
+	noWithdrawals := true
+
+	for _, alloc := range params.Allocations {
+		if alloc.Amount.IsNegative() {
+			return nil, RPCErrorf("negative allocation: %s for asset %s", alloc.Amount, alloc.AssetSymbol)
+		}
+
+		walletAddress := GetWalletBySigner(alloc.ParticipantWallet)
+		if walletAddress == "" {
+			walletAddress = alloc.ParticipantWallet
+		}
+
+		currentAmount := currentAllocations[walletAddress][alloc.AssetSymbol]
+		if alloc.Amount.LessThan(currentAmount) {
+			withdrawalAmount := currentAmount.Sub(alloc.Amount)
+			noWithdrawals = false
+
+			userAddress := common.HexToAddress(walletAddress)
+			userAccountID := NewAccountID(walletAddress)
+			ledger := GetWalletLedger(tx, userAddress)
+
+			if err := ledger.Record(sessionAccountID, alloc.AssetSymbol, withdrawalAmount.Neg()); err != nil {
+				return nil, RPCErrorf("failed to debit session account: %w", err)
+			}
+			if err := ledger.Record(userAccountID, alloc.AssetSymbol, withdrawalAmount); err != nil {
+				return nil, RPCErrorf("failed to credit user account: %w", err)
+			}
+			_, err = RecordLedgerTransaction(tx, TransactionTypeAppWithdrawal, sessionAccountID, userAccountID, alloc.AssetSymbol, withdrawalAmount)
+			if err != nil {
+				return nil, RPCErrorf("failed to record transaction: %w", err)
+			}
+
+			participants[walletAddress] = true
+		} else if alloc.Amount.GreaterThan(currentAmount) {
+			return nil, RPCErrorf("incorrect withdrawal request: increased allocation for participant %s", walletAddress)
+		}
+	}
+
+	if noWithdrawals {
+		return nil, RPCErrorf("incorrect withdrawal request: non-negative allocation sum delta")
```

Comment:
```suggestion
		return nil, RPCErrorf("incorrect withdrawal request: non-negative allocations sum delta")
```

Changing for consistency.

Reply:
d72cdb6836ce23fb6174889a00d3c157cf91cbf8